### ISSUE TITLE
[HELP] setName and assert

### ISSUE THREAD

**Issue by iRaziul** [Respect/Validation#1333]
This is Ok
![image](https://user-images.githubusercontent.com/51883557/106729675-f4ba0c00-6637-11eb-8949-0af7a8ca8089.png)

Output:
![image](https://user-images.githubusercontent.com/51883557/106729743-0a2f3600-6638-11eb-8ade-7165a3ef8c5a.png)

### But the the problem raises when I use `setName` method.
![image](https://user-images.githubusercontent.com/51883557/106730069-5f6b4780-6638-11eb-9781-248bf1fdced0.png)
Output (Only the last error message):
![image](https://user-images.githubusercontent.com/51883557/106730166-7b6ee900-6638-11eb-971e-ad743e65c2de.png)

But I expected the returned array to look like this:
![image](https://user-images.githubusercontent.com/51883557/106730632-c8eb5600-6638-11eb-917e-5e13f9da1f38.png)

_In version 1 it worked as I expected but the problem occurs on the latest version_

I don't know if there is any other way to do this.

**Comment by iRaziul** [comment]
This might be the reason behind it.

The code on line `160` at `library\Factory.php`
```
$id = $params['name'] = $validatable->getName();
```
By changing this to the below code provided me the expected result.

```
$params['name'] = $validatable->getName();
```

**Comment by henriquemoody** [comment]
We already have a distinction between `name` and `id` in versions 2.x. @alganet, where you thinking about having `setName()` always update the name, but not the ID of the rule? That does sound like a good plan, and it would be probably feasible with the changes I made in the `main` branch

**Comment by henriquemoody** [comment]
I've implemented this feature!

You can see this in [issue-1333.phpt](https://github.com/Respect/Validation/blob/176a8e6d7ef7b1611fef7716a213ff9c3a6ce291/tests/integration/issue-1333.phpt). I would appreciate some feedback!

In the `main` branch, the messages also include the `__root__` message, which could be a bit of an overhead, but it's easy to ignore since it has a standard name.

**Comment by henriquemoody** [comment]
This fix will be available in version 3.0.

 I'm not sure yet when I will release that version, it will probably take a couple of months.

Thank you for reporting! 🐼
